### ISSUE TITLE
kompose up connection refused minikube due to the port is 8443 instead of 6443 under k8s1.16.0

### ISSUE THREAD

**Issue by moluzhui** [kubernetes/minikube#7663]
**Steps to reproduce the issue:** 

1. curl -Lo minikube https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 \
  && chmod +x minikube
2. curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.16.0/bin/linux/amd64/kubectl
3.minikube start --driver=none --kubernetes-version v1.16.0

**Full output of failed command:** 
```
kompose -f docker-compose.yaml up
INFO We are going to create Kubernetes Deployments, Services and PersistentVolumeClaims for your Dockerized application. If you need different kind of resources, use the 'kompose convert' and 'kubectl create -f' commands instead. 
 
FATA Error while deploying application: Get https://127.0.0.1:6443/api: dial tcp 127.0.0.1:6443: connect: connection refused
```
I think this is due to a port mismatch,  Kompose requires a `6443` port, and the k8s created by minikube is a `8443` port instead of the usual 6443
```
kubectl cluster-info
Kubernetes master is running at https://172.26.90.122:8443
KubeDNS is running at https://172.26.90.122:8443/api/v1/namespaces/kube-system/services/kube-dns:dns/proxy

To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
```

How to **configure the minikube start command** to change the `port to 6443`?

**Full output of `minikube start` command used, if not already included:**
```
# minikube start --driver=none --kubernetes-version v1.16.0
😄  minikube v1.9.2 on Ubuntu 18.04
✨  Using the none driver based on user configuration
👍  Starting control plane node  in cluster minikube
🤹  Running on localhost (CPUs=16, Memory=62389MB, Disk=503837MB) ...
ℹ️  OS release is Ubuntu 18.04.3 LTS
🐳  Preparing Kubernetes v1.16.0 on Docker 18.09.7 ...
    ▪ kubelet.resolv-conf=/run/systemd/resolve/resolv.conf
❗  This bare metal machine is having trouble accessing https://k8s.gcr.io
💡  To pull new external images, you may need to configure a proxy: https://minikube.sigs.k8s.io/docs/reference/networking/proxy/
🌟  Enabling addons: default-storageclass, storage-provisioner
🤹  Configuring local host environment ...

❗  The 'none' driver is designed for experts who need to integrate with an existing VM
💡  Most users should use the newer 'docker' driver instead, which does not require root!
📘  For more information, see: https://minikube.sigs.k8s.io/docs/reference/drivers/none/

❗  kubectl and minikube configuration will be stored in /root
❗  To use kubectl or minikube commands as your own user, you may need to relocate them. For example, to overwrite your own settings, run:

    ▪ sudo mv /root/.kube /root/.minikube $HOME
    ▪ sudo chown -R $USER $HOME/.kube $HOME/.minikube

💡  This can also be done automatically by setting the env var CHANGE_MINIKUBE_NONE_USER=true
🏄  Done! kubectl is now configured to use "minikube"
💡  For best results, install kubectl: https://kubernetes.io/docs/tasks/tools/install-kubectl/
```

**Comment by afbjorklund** [comment]
See https://github.com/kubernetes/kompose/pull/1243 (v1.21.0), you can now use --server to choose your URL.

`      --server string       kubernetes apiserver url (default "https://127.0.0.1:6443")`

The default used to be http://localhost:8080, so it cannot really be trusted anyway.

Normally people don't work on the control plane, so have to supply an IP to use too...

**Comment by tstromberg** [comment]
@moluzhui - does this solution work for you?

**Comment by moluzhui** [comment]
@tstromberg I've solved this problem by adding `--apiserver-port=6443` in the `minikube start` command and changing the port to 6443.

I just tried to use --server https://127.0.0.1:8443 in kompose up. And as I expected the following error occurred because the k8s port was changed to 6443 by `minikube start --apiserver-port=6443`.
```
FATA Error while deploying application: Get https://127.0.0.1:8443/api: dial tcp 127.0.0.1:8443: connect: connection refused
```
This does seem to change the url and could be another solution for me.
